### DISCUSSION TITLE
GitHub webhook installation payload does not contain repositories array

### DISCUSSION THREAD

**Discussion by DominicRoyStang** [community/community#24524] [Apps, API and Webhooks]
<p>For context, I’m building a GitHub App, and keeping track of repositories on which the app is installed. I’m using the following webhooks library: <a href="https://www.npmjs.com/package/@octokit/webhooks" rel="nofollow noopener">https://www.npmjs.com/package/@octokit/webhooks</a></p>
<p>From what I understand, there are two webhooks that can add repositories:</p>
<ul>
<li><code>installation.created</code></li>
<li><code>installation_repositories.added</code></li>
</ul>
<p>For repository deletion, their counterparts would be</p>
<ul>
<li><code>installation.deleted</code></li>
<li><code>installation_repositories.removed</code></li>
</ul>
<p>In my testing, the payload does not contain a <code>repositories</code> array in the webhook event on <code>installation.deleted</code>. The spec clearly shows that it should be present: <a href="https://developer.github.com/webhooks/event-payloads/#webhook-payload-example-12" rel="nofollow noopener">https://developer.github.com/webhooks/event-payloads/#webhook-payload-example-12</a></p>
<p>I’m assuming that this is a bug, but I would like some confirmation. Maybe something is wrong on my end. If it is a bug, could someone point me to the correct location to report it?</p>
<details>
<p>
Screenshot from smee.io: no repositories key is present in the payload (there is also no repositories key under installation and sender)
</p>
<p></p>
<div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/1/12ba0e0e88891c4eb65c42cb69dce496fe428cb6.png" data-download-href="/uploads/short-url/2FFdlvwzi2wMg5tvgMxLttH0vuC.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/28553263/181085173-2699f610-0702-4499-8dca-c0d0f7443a1d.png" alt="image" data-base62-sha1="2FFdlvwzi2wMg5tvgMxLttH0vuC" width="690" height="270" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/1/12ba0e0e88891c4eb65c42cb69dce496fe428cb6_2_690x270.png, https://aws1.discourse-cdn.com/github/original/2X/1/12ba0e0e88891c4eb65c42cb69dce496fe428cb6.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/1/12ba0e0e88891c4eb65c42cb69dce496fe428cb6.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/1/12ba0e0e88891c4eb65c42cb69dce496fe428cb6_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">747×293 21.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Yet on <em>installation.created</em> and <em>installation_repositories.removed</em>, the correct repository* keys are present</p>

</details>

**Comment by francisfuzz** [comment]
<aside class="quote no-group" data-username="DominicRoyStang" data-post="1" data-topic="116562">
<div class="title">
<div class="quote-controls"></div>
 DominicRoyStang:</div>
<blockquote>
<p>In my testing, the payload does not contain a <code>repositories</code> array in the webhook event on <code>installation.deleted</code> . The spec clearly shows that it should be present: <a href="https://developer.github.com/webhooks/event-payloads/#webhook-payload-example-12">https://developer.github.com/webhooks/event-payloads/#webhook-payload-example-12 </a></p>
<p>I’m assuming that this is a bug, but I would like some confirmation. Maybe something is wrong on my end. If it is a bug, could someone point me to the correct location to report it?</p>
</blockquote>
</aside>
<p>:wave: <a class="mention" href="/dominicroystang">@DominicRoyStang</a> –– Thanks for reporting this! The Support Community Forum is a great place to ask these questions.</p>
<p>I checked on our and see that we’ve got an internal issue open for tracking this. Our engineers working on GitHub Apps are aware of this but we don’t have a timeline for when <code>repositories</code> will be included in that payload. However, we will update you again as soon as we have more information to share.</p>
<p>Thanks for your patience!</p>

**Comment by DominicRoyStang** [comment]
<p><a class="mention" href="/francisfuzz">@francisfuzz</a> thanks for the response, I’m glad it’s being worked on.</p>
<p>Looking forward to a fix in the future. Keep me posted!</p>

**Comment by francisfuzz** [comment, marked as answer]
<p>:wave: <a class="mention" href="/dominicroystang">@DominicRoyStang</a>––Great news! Our engineering team deployed an update to GitHub Apps webhook payloads recently, such that the <code>installation.deleted</code> event includes the installation’s repositories (consistent with our <a href="https://developer.github.com/webhooks/event-payloads/#installation">documentation for the <code>installation</code> event payload</a>). We (kindly) ask that you give this a try on your application and let us know if you’re still seeing the originally reported behavior. We hope this helps!</p>

**Comment by DominicRoyStang** [comment]
<p><a class="mention" href="/francisfuzz">@francisfuzz</a> I just tested it, and it works as expected now! Thank you for the update, and thanks to the engineering team for the fast resolution!</p>
